### PR TITLE
NFTBank integration tests

### PR DESCRIPTION
**Pull request by jnapprogs**
## Description
Integration test for NFTBank

### CODE REVIEW

File: packages/sources/heni/test/integration/fixtures.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const mockPriceResponseSuccessMalformed = (): nock =>
       'x-api-key': process.env.API_KEY,
     },
   })
-    .get(`/price`)
+    .post(`/price`)
```
Comment (vnavascues) [inline packages/sources/heni/test/integration/fixtures.ts]: Why this change? The `price` endpoint makes a `get` request and nock is incapable of intercepting the request:

```shell
{"level":"error","time":1646132310974,"pid":184977,"hostname":"victor-Vostro-7590","message":"AdapterError: Request failed with status code 500","jobRunID":"1","params":{"endpoint":"nftbank","nftCollection":0,"nftId":1,"pricingAsset":"ETH","resultPath":"nftbank"},"feedID":"{\"data\":{\"endpoint\":\"nftbank\",\"nftCollection\":0,\"nftId\":1,\"pricingAsset\":\"ETH\",\"resultPath\":\"nftbank\"}}","url":"https:/api.nftbank.ai/estimates-v2/estimates/0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d/1"}
```

File: packages/sources/nftbank/package.json [hunk 2]
```diff
@@ -36,6 +36,9 @@
     "@chainlink/types": "workspace:*",
     "@types/jest": "27.0.3",
     "@types/node": "16.11.19",
+    "@types/supertest": "2.0.11",
+    "nock": "^13.2.4",
```
Comment (vnavascues) [inline packages/sources/nftbank/package.json]: please remove the caret `^`, we do follow Chainlink practices and they remove them on `devDependiencies`

**Review by vnavascues — CHANGES_REQUESTED** [review]
Juan I was expecting to see the revamp (bugfixes + refactor) included in this PR, for instance:
- Better extensibility: `NFTCollection` as Map, don't use `getNftCollection()`, improved `getEstimateFor()`.
- Don't overwrite the real API result: don't use `createResultData()`.
- Support `0` price: don't use `Requester.validateResultNumber()` (throws an error), don't use `createResultData()` (sets `item_found` to false if `estimate` is falsy).
- Don't use `Requester.validateResultNumber()` just
- Improved errors: the error in  `getEstimateFor()` should be a simple `throw new Error("<message>")`. Then outside re-raise it as `AdapterError` with a 200 statusCode and including `cause` and `url` properties.
- Delete unused types, variables, etc.

I've added an extra commit with the expected changes. It's linted, and it builds, however I've not tested it!

File: packages/sources/heni/test/integration/fixtures.ts (hunk 1 above)
Comment (jnapprogs) [reply packages/sources/heni/test/integration/fixtures.ts]: Oh snap.  I did not mean to make this change.  I shall go and revert it.

**Comment by jnapprogs** [comment]
@vnavascues, I have renamed the branch to better align with the ticket.  I will go ahead and close this PR and reference the new PR in this comment

**Pull request closed without merging**